test/test_engine.js

```diff
@@ -1142,6 +1142,50 @@ function buildEngineTests() {
       const unitsAfterReplace2 = engine.getLastSpecifiedUnits("manufacture");
       assert.equal(unitsAfterReplace2, "units");
     });
+
+    QUnit.test("initial charge considers last specified units", function (assert) {
```

Comment:
Is it true that the two tests in here need to have recharge in order to validate the new logic?